template/classroom_setup.sh.erb

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+<%- if context.cluster =~ /kubernetes/ -%>
+exec &> >(tee -a "pod.log")
+<%- end -%>
+
+# Quit if any function returns an non-zero status.
+set -eE
+set -o pipefail
+
+project_id="<%= context.account %>"
+class_id="<%= context.classroom %>"
+classroom_version="<%= context.module_type %>"
+
+find_project_directory() {
+  ESS_DIR="/fs/ess/$project_id"
+  PROJ_DIR="/fs/project/$project_id"
+  if [[ -d $ESS_DIR ]]; then
+    echo $ESS_DIR
+    ls $ESS_DIR > /dev/null 2>&1
+    if [[ $? != 0 ]]; then
+      echo "Cannot access $ESS_DIR"
+      exit 1
+    fi
+  elif [[ -d $PROJ_DIR ]]; then
+    echo $PROJ_DIR
+    ls $PROJ_DIR > /dev/null 2>&1
+    if [[ $? != 0 ]]; then
+      echo "Cannot access $PROJ_DIR"
+      exit 1
+    fi
+  else
+    echo "Cannot find any project directory for {project_id}"
+    exit 1
+  fi
+}
+
+find_project_owner() {
+  project_dir=$(find_project_directory)
+  echo $(getfacl -t project_dir |& grep USER |awk '{print $2}')
+}
+
+create_classroom_directory() {
+  class_dir=$1
+
+  echo "Creating class directory $class_dir" 
+  umask 027
+  mkdir -v -p $class_dir
+  chgrp $project_id $class_dir
+  chmod g+s $class_dir
+  setfacl -R -d -m g:${project_id}:r-X $class_dir
+
+  echo "Creating materials directory $class_dir/materials"
+  mkdir -v -p $class_dir/materials
+
+  echo "Creating workspace directory $class_dir/workspace"
+  mkdir -v -p $class_dir/workspace
+}
+
+create_jupyter_classroom() {
+  project_dir="$1"
+  class_dir="$project_dir/$class_id"
+  jupyter_dir="$class_dir/jupyter"
+
+  if [[ -d $jupyter_dir ]]; then
+    echo "The Jupyter environment $jupyter_dir has been created"
+    return 0
+  fi
+
+  if [[ ! -d $class_dir ]]; then
+    create_classroom_directory $class_dir
+  fi
+
+  echo "Creating Jupyter environment with $(which python3)"
+  python3 -m venv --without-pip --prompt "$class_id" $jupyter_dir
+  source $jupyter_dir/bin/activate
+  # curl https://bootstrap.pypa.io/get-pip.py |python # RHEL9
+  curl https://bootstrap.pypa.io/pip/3.6/get-pip.py |python # RHEL7
```

Comment:
You seem to have both EL7 and EL9 here. If you need to toggle on the OS you can `source /etc/os-release` and toggle based off of some environment variables like `VERSION_ID`. 

Here's an example of the same in our visit application:



https://github.com/OSC/bc_osc_visit/blob/94ca22543b4b5c28e4157a4437d93ee248489783/template/script.sh.erb#L20-L32

Reply:
I created the RHEL7 line because I was testing on Pitzer RHEL7 while I did not yet have access to the RHEL9 classroom environment. We don’t need to toggle the system on; we only need the RHEL9 line in production.

Reply:
OK cool - that's what I did for my updates, just used all the EL9 stuff.